client/verta/verta/client.py

```diff
@@ -1199,7 +1197,15 @@ class Project(_ModelDBEntity):
         Experiment Runs under this Project.
 
     """
-    def __init__(self, conn, conf,
+    def __init__(self, conn, conf, id_):
+        super(Project, self).__init__(conn, conf, _ProjectService, "project", id_)
+
+        self._conn = conn
+
+        self.id = id_
+
+    @staticmethod  # TODO: if PR approved, move this fn down with other static/class methods
+    def _get_or_create(conn, conf,
```

Comment:
Here's where the logic is moved out of `Project.__init__()` and into `_get_or_create()`